Cleanup guiglobals

lenhard opened the pull request:
Implements #1408

Cleans up the `GUIGlobals` class and move `LatexFieldFormatter` to the `bibtex` package, which is now independent of the `gui` package.

As a final step, I would move the class `net.sf.jabref.importer.fileformat.FieldContentParser` to the `bibtex` package, which would make the package self-contained (apart from imports to the preferences and related classes) and would move the whole `bibtex` package into `logic`. Are there objections to this?

lenhard commented:
I restructured variable positioning as discussed in #1408 and turn MAX_BACK_HISTORY_SIZE, LINE_LENGTH, and INDENT into proper preferences.

@JabRef/developers What about turning the `bibtex` package into a subpackage of `logic`? Or should that rather stay a top-level package?

simonharrer commented:
The more stuff is in the logic package, the better.

lenhard commented:
Ok, from side this is good to go. I even was able to move some more classes into the model package.

matthiasgeiger commented:
LGTM

simonharrer commented on src/main/java/net/sf/jabref/JabRefPreferences.java:
```diff
@@ -75,12 +73,16 @@
 public class JabRefPreferences {
+    // Character separating field names that are to be used in sequence as
+    // fallbacks for a single column (e.g. "author/editor" to use editor where
+    // author is not set):
```
This comment is placed wrongly, I think.

simonharrer commented on src/test/java/net/sf/jabref/exporter/BibDatabaseWriterTest.java:
```diff
@@ -269,6 +269,7 @@ public void writeEntryWithCustomizedTypeAlsoWritesTypeDeclaration() throws IOExc
                         + Globals.NEWLINE + Globals.NEWLINE
                         + "@Comment{jabref-entrytype: Customizedtype: req[required] opt[optional]}" + Globals.NEWLINE,
                 stringWriter.toString());
+        EntryTypes.removeAllCustomEntryTypes();
```
Hm, maybe use a try-finally here instead?

tobiasdiez replied:
or in teardown method

simonharrer commented on src/main/java/net/sf/jabref/gui/maintable/MainTableColumn.java:
```diff
@@ -63,7 +64,7 @@ public String getDisplayName() {
         }
 
-        StringJoiner joiner = new StringJoiner(MainTableFormat.COL_DEFINITION_FIELD_SEPARATOR);
+        StringJoiner joiner = new StringJoiner(JabRefPreferences.COL_DEFINITION_FIELD_SEPARATOR);
```
I do not understand this. Why move a constant to JabRefPreferences which is not a preference?

simonharrer replied:
Another comment: the method getDisplayName is called quite often. If we would query a preference for each field in each row having 10k rows, this would lead to performance issues.

lenhard replied:
The code is wrong anyway and somehow slipped my attention, thanks for noticing! It should have been `Globals.prefs.getValue(JabRefPreferences.COL_DEFINITION_FIELD_SEPARATOR)` of course. 

Regarding the migration of constants into preferences: that is what *you* asked me to do in #1408 ;-) However, it is no hassle to move this particular constant back into `GUIGlobals`. 

Will do, as with the rest of your comments

tobiasdiez commented on src/main/java/net/sf/jabref/exporter/ExportCustomizationDialog.java:
```diff
@@ -61,16 +60,21 @@
 public class ExportCustomizationDialog extends JDialog {
 
+    // Column widths for export customization dialog table:
+    private static final int EXPORT_DIALOG_COL_1_WIDTH = 200;
```
Remove prefix `Export dialog` in the names?

tobiasdiez commented on src/main/java/net/sf/jabref/gui/SidePane.java:
```diff
@@ -33,7 +33,9 @@
 public class SidePane extends JPanel {
 
-    private final Dimension PREFERRED_SIZE = new Dimension(GUIGlobals.SPLIT_PANE_DIVIDER_LOCATION, 100);
+    private static final int SPLIT_PANE_DIVIDER_LOCATION = 175;
```
Not necessary as a separate variable and can be merged with next line, right?

tobiasdiez commented:
LGTM, just some minor remarks.

What I do not understand is the purpose of the bibtex package. Some of the classes seem like they do not have anything to do with writing or parsing bibtex and a rather general, for example the comparators. Maybe it would make sense to completely dissolve the bibtex package by moving the parsing logic to the import package and writing to export...in the end the bibtex format is just a special import-export format.

lenhard commented:
@tobiasdiez Thanks for your remarks, I will integrate them as well.

As for the bibtex package: Its existence is a historical thing and I do not know the exact motivation behind. It seems like "bundle parsing and writing of pure bibtex", but, with the addition of new importers and exporters, responsibilities shifted and mixed over time. The big advantage we have now is that the package moved to logic, which means that there is a guarantee that it is independent of GUI code and it can be re-used for new GUI stuff with relative safety. If we move it out of logic (to importers or exporters) we loose that advantage. Ultimately, the goal should be to get the GUI independent code from the import/export package into logic as well.

lenhard commented:
I implemented the above comments and the PR is once again ready to go from my side.

One remark: `COL_DEFINITION_FIELD_SEPARATOR` cannot go to `GUIGlobals`, since this would create dependencies from logic to gui. Therefore, I moved it into `COL_DEFINITION_FIELD_SEPARATOR` to `Globals`. This is fair, I think, since the variables relates to our BibTeX sytnax and field content, which is independent of the GUI.

I still don`t get how the coverage can have increased that much.
